src/server/scripts/Kalimdor/DireMaul/instance_dire_maul.cpp

```diff
@@ -60,8 +78,203 @@ class instance_dire_maul : public InstanceMapScript
     {
         instance_dire_maul_InstanceMapScript(Map* map) : InstanceScript(map)
         {
+            SetHeaders(DataHeader);
             SetBossNumber(EncounterCount);
         }
+
+        void OnCreatureCreate(Creature* creature) override
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_IMMOLTHAR:
+                    _immoGUID = creature->GetGUID();
+                    // we make Immolthar non attackable, otherwise players with pets can pull him out of the forcefield
```

Comment:
Really? Seems like now I understand the point of boundaries in places where Blizzard already added boundaries(that thing is called door)

Reply:
@Jinnaix could you please add a boundary here instead of flags? But only if he shouldn't have unnattackable flags

Reply:
Will try, wanted to finish the PR this weekend anyway.
Never did boundaries before but that shouldn't be witchcraft :) 
Im using unattackable here since our mmaps dont support door type gameobjects for pets/minions. Not the only place where this is problematic btw.

Reply:
Nvm, I forgot one thing, I have no idea if he should not be pulled out of that circle **after 5 pylons are done**, maybe in retail there's no boundary
Without info maybe it's better to hold that flag

Reply:
on retail the boss is targeteable, but you cannot send pet to attack it.

Reply:
PTR https://i.imgur.com/GQpzNsb.png https://i.imgur.com/4rt11mR.png

Reply:
Well, i would like to keep it this way with a note that as long as mmaps doesnt support doors, we keep this as a workaround?
Or should i completely leave this one out, so players with pets/minions can bypass the pylon event?

Reply:
Keep the flag